File: 69. Sqrt(x)/binarySearch.cpp
class Solution {
public:
    int mySqrt(int x) {
        int low = 0, high = x, res = 0;
        while (low <= high) {
            long long mid = low + (high - low) / 2;
            if (mid * mid <= x) {
                res = mid;
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }
        return res;
    }
};

File: 69. Sqrt(x)/binarySearch.py
class Solution:
    def mySqrt(self, x: int) -> int:
        low, high, res = 0, x, 0
        while low <= high:
            mid = (low + high) // 2
            if mid * mid <= x:
                res = mid
                low = mid + 1
            else:
                high = mid - 1
        return res